src/App.js

```diff
@@ -35,22 +35,36 @@ const App = () => {
       .catch((err) => console.log(err));
   }, []);
 
-  const handleToggleSubmit = (event) => {
-    setIsDaily(!isDaily);
+  const handleToggleDaily = (event) => {
+    const target = event.target;
+    const targetIsDaily = target.classList.contains('daily');
+    const targetIsWeekly = target.classList.contains('weekly');
+    if ((targetIsDaily && !isDaily) || (targetIsWeekly && isDaily)) {
+      setIsDaily(!isDaily);
+    }
   }
 
   return (
     <section className='app'>
       <Sidebar />
       <section className='display'>
         <Route exact path='/'>
-          <Button
-            className="toggle-daily"
-            onClick={handleToggleSubmit}
-            color={isDaily ? "primary" : "secondary"}
-          >
-              {isDaily ? "Daily" : "Weekly"}
-          </Button>
+          <div className="toggle-daily" role="button">
+            <Button
+              className="daily"
+              onClick={handleToggleDaily}
```

Comment:
I would modify this to pass through your daily and weekly argument directly into handleToggleDaily to avoid directly targeting the DOM. You can do that like this:

`onClick={() => handleToggleSubmit(false)}`

If that was for a specific design reason let me know and we can discuss.

const handleToggleDaily = (toggleValue) => {

Reply:
You're right. This can definitely be done without targeting the DOM. I refactored to only use the component's state.

I also moved the button group to the Home component, as you suggested.